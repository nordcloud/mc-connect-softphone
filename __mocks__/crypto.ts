export default {
  randomBytes: jest.fn().mockReturnValue({
    toString: jest.fn().mockReturnValue('dummyRandomBytesToString'),
  }),

  createHash: jest.fn().mockReturnValue({
    update: jest.fn().mockReturnValue({
      digest: jest.fn().mockReturnValue('dummyCreateHashUpdateDigest'),
    }),
  }),
  randomUUID: jest.fn().mockReturnValue('dummyUUID'),
};
