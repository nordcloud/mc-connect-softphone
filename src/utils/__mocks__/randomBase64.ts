export const randomBase64 = jest.fn().mockResolvedValue('randomBase64Value');
