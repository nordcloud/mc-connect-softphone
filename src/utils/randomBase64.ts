import crypto from 'crypto';
import { promisify } from 'util';

export async function randomBase64(size: number) {
  const cryptoRandomBytes = promisify(crypto.randomBytes);
  const bytes = await cryptoRandomBytes(size);

  return bytes.toString('base64url');
}
