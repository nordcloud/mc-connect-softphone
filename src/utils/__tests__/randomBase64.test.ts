import crypto from 'crypto';

import { randomBase64 } from '../randomBase64';

jest.mock('util');
jest.mock('crypto');

const arg = 64;

describe('randomBase64', () => {
  it('calls crypto.randomBytes with correct args', async () => {
    await randomBase64(arg);
    expect(jest.mocked(crypto.randomBytes).mock.calls).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    expect(randomBase64(arg)).resolves.toMatchSnapshot();
  });
});
