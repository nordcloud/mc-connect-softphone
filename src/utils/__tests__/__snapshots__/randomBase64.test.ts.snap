// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`randomBase64 calls crypto.randomBytes with correct args 1`] = `
[
  [
    64,
  ],
]
`;

exports[`randomBase64 resolves with a correct value 1`] = `"dummyRandomBytesToString"`;
